packages/compass/src/app/components/home.tsx

```diff
@@ -161,51 +135,49 @@ function reducer(state: State, action: Action): State {
   }
 }
 
-function showCollectionSubMenu({ isReadOnly }: { isReadOnly: boolean }) {
-  void hadronIpc.ipcRenderer?.call('window:show-collection-submenu', {
-    isReadOnly,
+async function reauthenticationHandler() {
+  const confirmed = await showConfirmation({
+    title: 'Authentication expired',
+    description:
+      'You need to re-authenticate to the database in order to continue.',
   });
+  if (!confirmed) {
+    throw new Error('Reauthentication declined by user');
+  }
 }
 
-function hideCollectionSubMenu() {
-  void hadronIpc.ipcRenderer?.call('window:hide-collection-submenu');
-}
-
-function notifyMainProcessOfDisconnect() {
-  void hadronIpc.ipcRenderer?.call('compass:disconnected');
-}
+export type HomeProps = {
+  appName: string;
+  getAutoConnectInfo?: () => Promise<ConnectionInfo | undefined>;
+  showWelcomeModal?: boolean;
+  createFileInputBackend: () => FileInputBackend;
+  onDisconnect: () => void;
+  showCollectionSubMenu: (args: { isReadOnly: boolean }) => void;
+  hideCollectionSubMenu: () => void;
+  showSettings: () => void;
+  __TEST_MONGODB_DATA_SERVICE_CONNECT_FN?: () => Promise<DataService>;
+  __TEST_CONNECTION_STORAGE?: typeof ConnectionStorage;
+};
 
 function Home({
   appName,
   getAutoConnectInfo,
-  isWelcomeModalOpenByDefault = false,
+  showWelcomeModal = false,
+  createFileInputBackend,
+  onDisconnect,
+  showCollectionSubMenu,
+  hideCollectionSubMenu,
```

Comment:
Small nit: I'd rather delegate this outside of the component by exposing onWorkspaceChange instead of passing as actions. First of all, passing actions is not really a React pattern (but this is more of a naming thing and communication of what controls the logic), second this makes the interface closer to compass-web, which might eventually allow us to unify more parts of the shared code